Documentation/observability/visibility.rst

```diff
@@ -12,63 +12,72 @@ Layer 7 Protocol Visibility
 
 .. note::
 
-    This feature requires enabling L7 Proxy support. Without it, the visibility annotation is ignored.
+    This feature requires enabling L7 Proxy support.
 
-While :ref:`monitor` provides introspection into datapath state, by default it
-will only provide visibility into L3/L4 packet events. If :ref:`l7_policy` are
-configured, one can get visibility into L7 protocols, but this requires the full
-policy for each selected endpoint to be written. To get more visibility into the
-application without configuring a full policy, Cilium provides a means of
-prescribing visibility via `annotations <https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/>`_
-when running in tandem with Kubernetes.
+While :ref:`monitor` provides introspection into datapath state, by default, it
+will only provide visibility into L3/L4 packet events. If you want L7
+protocol visibility, you can use L7 Cilium Network Policies (see :ref:`l7_policy`).
 
-Visibility information is represented by a comma-separated list of tuples in
-the annotation:
 
-``<{Traffic Direction}/{L4 Port}/{L4 Protocol}/{L7 Protocol}>``
-
-For example:
-
-::
-
-  <Egress/53/UDP/DNS>,<Egress/80/TCP/HTTP>
-
-
-To do this, you can provide the annotation in your Kubernetes YAMLs, or via the
-command line, e.g.:
-
-.. code-block:: shell-session
-
-    kubectl annotate pod foo -n bar policy.cilium.io/proxy-visibility="<Egress/53/UDP/DNS>,<Egress/80/TCP/HTTP>"
+.. note::
 
-Cilium will pick up that pods have received these annotations, and will
-transparently redirect traffic to the proxy such that the output of
-``cilium monitor`` shows traffic being redirected to the proxy, e.g.:
+    Historically, it had been possible to enable L7 visibility using Pod
+    annotations (``policy.cilium.io/proxy-visibility``). This method is
+    no longer supported and we recommend users to switch to L7 policies instead.
+
+To enable visibility for L7 traffic, create a ``CiliumNetworkPolicy`` that specifies
```

Comment:
I think it's critical to make it crystal clear to users that this is not just a visibility policy, this is allowing traffic matching the statements below to flow. That was previously the difference between visibility policy vs this; visibility did not change the actual policy behaviour of network flows.

Reply:
this is why I had `toEntities: [ all ]`  at the end of the policy in the beginning -- to make the example behave more like a visibility policy. @joestringer do you think it's worth adding that catch-all rule back or just stress the point that this is a network policy that will drop traffic in the text?

Reply:
I think my feedback is less about the policy and more about having a dedicated paragraph  of test to frame the solution. My thought is that when users scan through the doc, they see that the position of the project is "we don't have a dedicated visibility-only solution, the way to solve this problem is with enforcement network policy and hence you need to do this with awareness of the security impacts".

Reply:
yep, makes sense. I've added a mention of this to a couple of places in the text

Reply:
LGTM thanks